test/knexfile.js

```diff
@@ -142,7 +152,7 @@ var testConfigs = {
     connection: testConfig.sqlite3 || {
       filename: __dirname + '/test.sqlite3'
     },
-    pool: pool,
+    pool: Object.assign({max: 1}, pool),
```

Comment:
@elhigu the sqlite3 tests were failing due to resource contention (multiple connections to the db file) on this branch w/o restricting the max pool size to 1. I'm not sure why the tests were passing on master, though b/c it seems sensible that the max pool size would have to be 1 for sqlite. Did we introduce a bug for sqlite3 in this PR for which the workaround is setting max pool size to 1?

Reply:
I'm not sure... I believe that nowadays sqlite also supports multiple connections. Which test was failing with bigger pool size?

Reply:
With `max: 1`, the tests always pass. Without it, I get two different failures, depending on whether `test/test.sqlite3` exists...

`$ rm test/test.sqlite3; DB=sqlite3 npm test` gives:

```
    knex.migrate (transactions disabled)
      knex.migrate.latest (all transactions disabled)
Knex:warning - migrations failed with error: SELECT foo FROM unknown_table - SQLITE_ERROR: no such table: unknown_table
        ✓ should create column even in invalid migration
      knex.migrate.latest (per-migration transaction disabled)
Knex:warning - migrations failed with error: SELECT foo FROM unknown_table - SQLITE_ERROR: no such table: unknown_table
        ✓ should run all working migration files in the specified directory
        1) should create column in invalid migration with transaction disabled


  758 passing (7s)
  1 failing

  1) sqlite3 | sqlite3 knex.migrate (transactions disabled) knex.migrate.latest (per-migration transaction disabled) should create column in invalid migration with transaction disabled:

      AssertionError: expected false to equal true
      + expected - actual

      -false
      +true

      at test/integration/migrate/index.js:300:29
  From previous event:
      at SchemaBuilder.Target.then (lib/interface.js:9:1751)
      at Context.<anonymous> (test/integration/migrate/index.js:299:77)
```

If I leave `test/test.sqlite3` existing, `DB=sqlite3 npm test` gives:

```
  sqlite3 | sqlite3
    ✓ #test number mssql should not allow unsafe bigint
    ✓ #test number mssql should allow safe bigint
    ✓ #1781 - decimal value must not be converted to integer
    Schema
      ✓ #1430 - .primary() & .dropPrimary() same for all dialects
      dropTable
        ✓ has a dropTableIfExists method
      createTable
        ✓ Callback function must be supplied
        1) is possible to chain .catch


  711 passing (2s)
  1 failing

  1) sqlite3 | sqlite3 Schema createTable is possible to chain .catch:
     create table `catch_test` (`id` integer not null primary key autoincrement) - SQLITE_ERROR: table `catch_test` already exists
  Error: SQLITE_ERROR: table `catch_test` already exists
      at Error (native)
  From previous event:
      at Client_SQLite3._query (lib/dialects/sqlite3/index.js:9:7581)
      at Client_SQLite3.query (lib/client.js:9:11871)
      at Runner.<anonymous> (lib/runner.js:9:7046)
  From previous event:
      at Runner.queryArray (lib/runner.js:9:9562)
      at lib/runner.js:9:3550
  From previous event:
      at Runner.run (lib/runner.js:9:2779)
      at SchemaBuilder.Target.then (lib/interface.js:9:1687)
      at SchemaBuilder.Target.(anonymous function) [as catch] (lib/interface.js:9:4430)
      at Context.<anonymous> (test/integration/schema/index.js:68:19)
```

Reply:
I think I figured it out - the sqlite3 dialect defines a `poolDefaults` method that sets `max: 1`, but that method is no longer called by client with the changes on this branch. I'll make another pass to figure out the fix.

Reply:
Sounds right, indeed I remember that at some point default for sqlite was set to that (some years ago).